tests/integration/shared-libs/cht-datasource/place.spec.js

```diff
@@ -0,0 +1,187 @@
+const { expect } = require('chai');
+const userFactory = require('@factories/cht/users/users');
+const {setAuth, removeAuth} = require('./auth');
+
+describe('cht-datasource Place', () => {
+  const contact0 = utils.deepFreeze(personFactory.build({ name: 'contact0', role: 'chw' }));
+  const contact1 = utils.deepFreeze(personFactory.build({ name: 'contact0', role: 'chw_supervisor' }));
+  const contact2 = utils.deepFreeze(personFactory.build({ name: 'contact0', role: 'program_officer' }));
+  const placeMap = utils.deepFreeze(placeFactory.generateHierarchy());
+  const place1 = utils.deepFreeze({ ...placeMap.get('health_center'), contact: { _id: contact1._id } });
+  const place2 = utils.deepFreeze({ ...placeMap.get('district_hospital'), contact: { _id: contact2._id } });
+  const place0 = utils.deepFreeze({
+    ...placeMap.get('clinic'),
+    contact: { _id: contact0._id },
+    parent: {
+      _id: place1._id,
+      parent: {
+        _id: place2._id
+      }
+    },
+  });
+  const placeType = 'clinic';
+  const clinic1 = utils.deepFreeze(placeFactory.place().build({
+    parent: {
+      _id: place1._id,
+      parent: {
+        _id: place2._id
+      }
+    },
+    type: placeType,
+    contact: {}
+  }));
+  const clinic2 = utils.deepFreeze(placeFactory.place().build({
+    parent: {
+      _id: place1._id,
+      parent: {
+        _id: place2._id
+      }
+    },
+    type: placeType,
+    contact: {}
+  }));
+
+  const userNoPerms = utils.deepFreeze(userFactory.build({
+    username: 'online-no-perms',
+    place: place1._id,
+    contact: {
+      _id: 'fixture:user:online-no-perms',
+      name: 'Online User',
+    },
+    roles: ['mm-online']
+  }));
+  const offlineUser = utils.deepFreeze(userFactory.build({
+    username: 'offline-has-perms',
+    place: place0._id,
+    contact: {
+      _id: 'fixture:user:offline-has-perms',
+      name: 'Offline User',
+    },
+    roles: ['chw']
+  }));
+  const dataContext = getRemoteDataContext(utils.getOrigin());
+  const expectedPlaces = [place0, clinic1, clinic2];
+
+  before(async () => {
+    setAuth();
+    await utils.saveDocs([contact0, contact1, contact2, place0, place1, place2, clinic1, clinic2]);
+    await utils.createUsers([userNoPerms, offlineUser]);
+  });
+
+  after(async () => {
+    await utils.revertDb([], true);
+    await utils.deleteUsers([userNoPerms, offlineUser]);
+    removeAuth();
+  });
+
+  describe('v1', () => {
+    describe('get', async () => {
+      const getPlace = Place.v1.get(dataContext);
+      const getPlaceWithLineage = Place.v1.getWithLineage(dataContext);
+
+      it('returns the place matching the provided UUID', async () => {
+        const place = await getPlace(Qualifier.byUuid(place0._id));
+        expect(place).excluding([ '_rev', 'reported_date' ]).to.deep.equal(place0);
+      });
+
+      it('returns the place with lineage when the withLineage query parameter is provided', async () => {
```

Comment:
suggestion:  Can you also call `getWithLineage` for the lowest place in the hierarchy when that place doesn't have a primary contact associated, so we make sure that this lineage-less call succeeds.